src/main.rs

```diff
@@ -101,34 +108,46 @@ fn main() -> ! {
     info!("Startup tc2-firmware {}", FIRMWARE_VERSION);
 
     // TODO: Check wake_en and sleep_en registers to make sure we're not enabling any clocks we don't need.
-    let mut peripherals = Peripherals::take().unwrap();
+    let mut peripherals: Peripherals = Peripherals::take().unwrap();
+    let is_audio = read_is_audio_from_rp2040_flash();
+
+    let freq;
+
+    if is_audio {
+        freq = ROSC_TARGET_CLOCK_FREQ_HZ_AUDIO.Hz();
+    } else {
+        //for some reason audio comes out faster than expected when using this clock
+        freq = ROSC_TARGET_CLOCK_FREQ_HZ_THERMAL.Hz();
+    }
     let (clocks, rosc) = clock_utils::setup_rosc_as_system_clock(
         peripherals.CLOCKS,
         peripherals.XOSC,
         peripherals.ROSC,
-        ROSC_TARGET_CLOCK_FREQ_HZ.Hz(),
+        freq,
     );
     let clocks: &'static ClocksManager = unsafe { extend_lifetime_generic(&clocks) };
+
     let system_clock_freq = clocks.system_clock.freq().to_Hz();
+
     info!(
         "System clock speed {}MHz",
         clocks.system_clock.freq().to_MHz()
     );
+
     // Watchdog ticks are required to run the timer peripheral, since they're shared between both.
+
     let mut watchdog = bsp::hal::Watchdog::new(peripherals.WATCHDOG);
     watchdog.enable_tick_generation((system_clock_freq / 1_000_000) as u8);
+
     watchdog.pause_on_debug(true);
     watchdog.start(8388607.micros());
+
     info!("Enabled watchdog timer");
-    let timer = bsp::hal::Timer::new(peripherals.TIMER, &mut peripherals.RESETS, clocks);
+    let timer = bsp::hal::Timer::new(peripherals.TIMER, &mut peripherals.RESETS, &clocks);
```

Comment:
Not sure why you added an extra reference to `clocks`?  It's already a ref.

Reply:
no idea i imagine i messed something up and vs code suggested it :( fixed now